.github/workflows/root-ci.yml

```diff
@@ -102,6 +107,9 @@ jobs:
           - platform: mac-beta
             arch: ARM64
             overrides: ["LLVM_ENABLE_ASSERTIONS=On", "CMAKE_CXX_STANDARD=20"]
+        exclude:
+          - event_name: push
+            platform: mac12
```

Comment:
You probably need my creative code here:
```
        # We have to get a bit creative here: GitHub actions only allows to
        # exclude partial matches, so we artificially add the event_name as
        # a "constant variable" that we can use to remove the Debug entries
        # for pull requests and on branch pushes. This is further complicated
        # by the fact that event_name is a string, but we need an array. So
        # we construct a JSON string that we can then convert into an array.
        event_name: ${{ fromJSON(format('["{0}"]', github.event_name)) }}
```

Also I don't agree this is what we want: I'd rather have `mac12` build on push and disable it in PRs, which are run more frequently.